### PR TITLE
[Bug 1538960] Correct ability to override openshift_management_app_template

### PR DESCRIPTION
**Pull request by mtnbikenc**
* Converted 'flavor' logic to be based on app template
* Converted 'ext-db' logic to use var based on app template
* Prepended role private vars with '__' for consistency

Bug 1538960
https://bugzilla.redhat.com/show_bug.cgi?id=1538960

### CODE REVIEW

**Comment by sdodson** [comment]
Nice work, thanks!
/lgtm

**Comment by mtnbikenc** [comment]
Updated: typo fix in vars/main.yml

File: roles/openshift_management/vars/main.yml [hunk 1]
```diff
@@ -30,14 +30,18 @@ __openshift_management_db_parameters:
...
+    short: miq
+    long: manageiq
+  cfme:
+    short: cfme
+    long: cloudforms
+
+__openshift_management_flavor: "{{ __openshift_management_flavors[openshift_management_app_template.split('-')[0]]['long'] }}"
+__openshift_management_flavor_short: "{{ __openshift_management_flavors[openshift_management_app_template.split('-')[0]]['short'] }}"
+
+__openshift_management_use_ext_db: "{{ true if 'ext-db' in openshift_management_app_template else false }}"
```
Comment (tbielawa) [inline roles/openshift_management/vars/main.yml]: smarter than what I was doing before :+1:

File: roles/openshift_management/vars/main.yml [hunk 2]
```diff
@@ -30,14 +30,18 @@ __openshift_management_db_parameters:
...
 
-# # Commented out until we can support both CFME and MIQ
-# # openshift_management_flavor: "{{ 'cloudforms' if openshift_deployment_type == 'openshift-enterprise' else 'manageiq' }}"
-#openshift_management_flavor: cloudforms
-openshift_management_flavor: manageiq
-# TODO: Make this conditional as well based on the prior variable
-# # openshift_management_flavor_short: "{{ 'cfme' if openshift_deployment_type == 'openshift-enterprise' else 'miq' }}"
-# openshift_management_flavor_short: cfme
-openshift_management_flavor_short: miq
+__openshift_management_flavors:
```
Comment (tbielawa) [inline roles/openshift_management/vars/main.yml]: I <3 what you did with this hash

**Pull request merged**